### DISCUSSION TITLE
Packaging the RML-HTML files

### DISCUSSION THREAD

**Discussion by TheParadox** [mikke89/RmlUi#299] [General]
Greetings, 

So, been playing around with this library and so far, like the looks of stuff. One question I have - mainly cause I have not used an HTML-based UI before is: packaging.

Is there a way to package the HTML/CSS files with the project .exe (I mean without hard coding the stuff, which in my opinion defeats the purpose...) or do they just need to be copied (or compressed and included with the project).

I think it would be cool for some aspects to allow the users (with enough knowledge) to customize their UI. But on the other end, trying to determine how best to ensure the files are there (regular text, compressed, or embedded).

Thanks for your thoughts

**Comment by mikke89** [comment]
Hello, appreciate the kind words about the library :)

The HTML/CSS files are really no different than any other asset you want to distribute with your application. It really depends on your goals and priorities.

Since you mention allowing users to customize it, I would suggest just leaving the source files next to the application. This is surely the simplest way for people to discover and play with it. You can see this is what we do if you download our sample release executables, then you can easily play with the RML/RCSS. You can also go an extra step and implement live reloading.

The next steps are zipping up collection of files or creating your own binary asset file types. These approaches may result in better control over loading and load times, and can be useful for versioning or packaging mod collections that users can install. Of course these steps are increasingly harder to play with for users, but I'm sure the most determined ones will figure it out and especially if you publish some guidelines.

What you also might want is a virtual file system so you can load assets from different packages (or the file system) based on some priority. This way users can eg. extract some files and play with them directly on the file system, and then package them up. I'm sure there are plenty of resources online if you want to go in this direction.

**Reply by TheParadox** [reply]
Awesome! Thank you - that is what I suspected, but wanted to verify before starting to head down that storyboard path.

Yeah - this library has been probably the easiest UI library, in terms of getting it working with any of my applications that I have worked with (minus TGUI -- but that is specifically made for SFML so it doesn't count.)

Thus, I may be posting a project or two in the next few months that use this UI :D 

Thanks again for the work on it!

**Reply by mikke89** [reply]
Very cool, looking forward to see the results!
